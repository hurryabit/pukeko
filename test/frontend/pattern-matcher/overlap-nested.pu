data Pair a b = | Pair a b
data Option a =
  | None
  | Some a
f :: Pair (Option a) a -> a
f xy =
  case xy of
  | Pair None     y -> y
  | Pair (Some x) _ -> x
  | Pair (Some _) y -> y
