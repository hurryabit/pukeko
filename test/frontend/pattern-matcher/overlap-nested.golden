test/frontend/pattern-matcher/overlap-nested.pu:5:3: overlapping patterns
